src/AppInstallerRepositoryCore/CompositeSource.h

```diff
@@ -41,7 +41,7 @@ namespace AppInstaller::Repository
         void AddAvailableSource(std::shared_ptr<ISource> source);
 
         // Sets the installed source to be composited.
-        void SetInstalledSource(std::shared_ptr<ISource> source);
+        void SetInstalledSource(std::shared_ptr<ISource> source, bool keepInstalledOnly = true);

```

Comment:
This seems unnecessary, as one can just filter out packages with no available versions.  Is that what you are using it for?

Reply:
This argument is to keep the existing behavior for list and upgrade (return only installed; filter out available not-installed packages) while extending it to allow me to get all the packages whether installed or not. For import we need all those packages here so we can do all the cases of "this is already installed, nothing to do", "this is already installed but can be updated" and "this is not yet installed".

Reply:
Ok, I think the name is confusing then.  I thought it meant:
```
true == results would include all installed packages
false == results would only include packages that are installed and correlated to an available package
```

I think it should be something more like `includeAllAvailablePackagesInSearchResults` or use an enum instead to drive the intended behavior:
```
// inside CompositeSource
enum class SearchBehavior
{
   AllInstalled,
   AllPackages,
   // More in the future maybe
};
```